hw/ip/kmac/data/kmac_base_testplan.hjson

```diff
@@ -158,4 +158,103 @@
       tests: ["{name}_throughput"]
     }
   ]
+
+  covergroups: [
+    {
+      name: config_cg
+      desc: '''
+            Covers that all valid configuration settings for the KMAC have been tested.
+            '''
+    }
+    {
+      name: msg_len_cg
+      desc: '''
+            Covers various input message length ranges, to ensure that KMAC can operate successfully
+            on different sized messages.
+            '''
+    }
+    {
+      name: digest_len_cg
+      desc: '''
+            Covers  various output digest length ranges, to ensure that KMAC can successfully output
+            various digest lengths.
+            Note that this only applies to XOF functions, as SHA3 functions have a fixed output
+            length.
+            '''
+    }
+    {
+      name: prefix_range_cg
+      desc: '''
+            The prefix used for CSHAKE and KMAC (function_name + customization_string) are only
+            allowed to be valid alphabet letters, or a space character.
+            This covergroup covers that all of these characters have appeared in a prefix value.
+            '''
+    }
+    {
+      name: msgfifo_write_mask_cg
+      desc: '''
+            Covers that the msgfifo has been written using all possible TLUL masks.
+            '''
+    }
+    {
+      name: msgfifo_status_cg
+      desc: '''
+            Covers that all possible fifo statuses have been seen in various operating states
```

Comment:
Perhaps enumerate the operating states here which will be crossed with the fifo levels. Also 'level' is more apt than 'status'?

Reply:
done, thanks!